tests/read.lisp

```diff
@@ -55,3 +55,101 @@
 
 (let ((*features* '(foo)))
   (test (= (read-from-string "#+foo 1 2)") 1)))
+
+;;; sharp radix reader
+
+(test (string= "this is 1985" (format nil "this is ~a" #o3701)))
+
+(test
+ (equal '((1 2573) (1 2) (1 2573) (1 2))
+        (let ((fn0 (lambda (x &optional (y #x0a0d)) (list x y) ))
+              (fn1 (lambda (x &key (y #x0a0d)) (list x y))))
+          (list
+           (funcall fn0 1)
+           (funcall fn0 1 2)
+           (funcall fn1 1)
+           (funcall fn1 1 :y 2)))))
+
+;;; radix sharp at sequence's
+
+#+jscl
+(test
+ (equal
+  '(t t t t t)
+  (let ((bfv #(#b1 #b10 #b11 #b100 #b101 #b110 #b111 #b1000 #b1001 #b1010))
+        (bfv1 #(#b1 #b10 #b11 'sharp #\# #b100 #b101 #b110 #b111 #b1000 #b1001 #b1010))
+        (bfv2 #(1 2 3 (QUOTE SHARP) #\# 4 5 6 7 8 9 10))
+        (xfv #(#xa001 #xb001 #xc001 #xd001 #xe001 #xf001)))
+    (list (equal
+           ;; correct b->d conversion
+           (jscl::vector-to-list bfv)
+           '(1 2 3 4 5 6 7 8 9 10))
+          (equal
+           ;; correct x->d conversion
+           (jscl::vector-to-list xfv)
+           '(40961 45057 49153 53249 57345 61441))
+          ;; list & sharp tokens -> with sharp reader
+          (equal
+           (jscl::vector-to-list bfv1)
+           '(1 2 3 (QUOTE SHARP) #\# 4 5 6 7 8 9 10))
+          (equal
+           (aref bfv1 3)
+           '(quote sharp))
+          (equal
+           (aref bfv1 4) #\#)  ))))
+
+;;; sharp radix reader from string
+;;; higly likely it redundant but let stay
+#+jscl
+(test
+ (equal
+  '(t)
+  (let* 
+      ((s1 (read-from-string "#(#b1 #b10 #b11 #b100 #b101 #b110 #b111 #b1000 #b1001 #b1010)")))
+    (list
+     (equal
+      (jscl::vector-to-list s1)
+      '(1 2 3 4 5 6 7 8 9 10))))))
+
+;;;
+;;; parse-integer
+;;;
+(test
+ (equal '(t t t t t t)
+        (list
+         (multiple-value-bind (num pos) (parse-integer " 11111000001 " :radix 2 )
+           (equal (list num pos) '(1985 13)))
+         (multiple-value-bind (num pos) (parse-integer " 7C1 " :radix 16 )
+           (equal (list num pos) '(1985 5)))
+         (multiple-value-bind (num pos) (parse-integer " 3701 " :radix 8 )
+           (equal (list num pos) '(1985 6)))
+         ;;clhs examples
+         (multiple-value-bind (num pos) (parse-integer "123")
+           (equal (list num pos) '(123 3)))
+         (multiple-value-bind (num pos) (parse-integer "123"  :start 1 :radix 5)
+           (equal (list num pos) '(13 3)))
+         (multiple-value-bind (num pos) (parse-integer "no-integer" :junk-allowed t)
+           (equal (list num pos) '(nil 0))))))
+
+;;;
+;;; other fun glitch's
+;;;
+
+#|
+If you remove comments from the following expression there will be a compilation error.
+actually any errors in the types are caught at the compilation stage.
+The correct value can be used in any expressions, as is, at your discretion
+|#
```

Comment:
🙂 interesting! But this is correct, right?

Reply:
Yes.

I removed comments.  Result  from `(bootstrap)` followed:

```
Compiling tests/package.lisp...
Compiling tests/print.lisp...
Compiling tests/read.lisp...

#x: bad base(16) digit at "ag"
   [Condition of type SIMPLE-ERROR]

Restarts:
 0: [RETRY] Retry SLIME REPL evaluation request.
 1: [*ABORT] Return to SLIME's top level.
 2: [ABORT] abort thread (#<THREAD "repl-thread" RUNNING {1003049D73}>)

Backtrace:
  0: (JSCL::SHARP-RADIX-READER #\x #<SB-IMPL::STRING-INPUT-STREAM {100565F1D3}>)
  1: (JSCL::LS-READ #<SB-IMPL::STRING-INPUT-STREAM {100565F1D3}> NIL #:G9904 T)
; Evaluation aborted on #<SIMPLE-ERROR "#~c: bad base(~d) digit at ~s~%" {1005777693}>.
```
On the JSCL console (REPL), it looks like this:
```
CL-USER> (let ((fn  
...         (lambda (#xag #xaf) (list #xaa #xaf)))) 
...   (funcall fn 1 2)) 
... 
ERROR: #x: bad base(16) digit at "ag"

CL-USER> 
```